apollo-router/src/json_ext.rs

```diff
@@ -26,7 +26,7 @@ pub(crate) type Object = Map<ByteString, Value>;
 const FRAGMENT_PREFIX: &str = "... on ";
 
 static TYPE_CONDITIONS_REGEX: Lazy<Regex> = Lazy::new(|| {
-    Regex::new(r"(?:\|\[(?<condition>.+)?\])")
+    Regex::new(r"\|\[(?<condition>.*?)\]")
```

Comment:
This will match `|[]` as a list with one item of empty string.
I think `"(?:\|\[(?<condition>.+?)?\])` is works better, but None in `extract_matched_conditions) should be interpreted as an empty vec.

Reply:
I was about to handle the empty string in `extract_matched_conditions()`, but I think your suggestion here is cleaner, pushed up the change in the [fc894e1](https://github.com/apollographql/router/pull/6431/commits/fc894e1c682449cad4b1e1adf69c0da31a843311). I've also updated `extract_matched_conditions()` to not return an `Option`.